drracket-tool-text-lib/drracket/private/syncheck/contract-traversal.rkt

```diff
@@ -210,7 +218,7 @@
   (let loop ([val (syntax-property stx prop)])
     (cond
       [(symbol? val)
-       (hash-set! map val (cons stx (hash-ref map val '())))]
+       (hash-update! map val (λ (v) (cons stx v)) '())]
```

Comment:
This one doesn't seem like an improvement. Not making it worse either, mind you, I suppose.

Reply:
This rule makes a much bigger difference when the code put into the lambda is larger, or when it's a single unary function that can be passed to `hash-update!` directly without a lambda at all. But this instance is a weird in-between case where the update code is extremely short (just `(cons stx v)`) but _isn't_ a single unary function call. Perhaps the rule could somehow skip cases like this, if it comes up often in practice. I'll keep an eye out and give it some thought.

Reply:
Thanks.